kernel/src/main/java/org/kframework/main/Main.java

```diff
@@ -64,17 +64,28 @@ public static boolean isNailgun() {
         return isNailgun;
     }
 
+    private static NGContext context = null;
+
+    public static void exit(int exitCode) {
+      if (context != null) {
+          context.exit(exitCode);
+      } else {
+          System.exit(exitCode);
+      }
+    }
+
     public static void nailMain(NGContext context) {
         long startTime = System.nanoTime();
         KServerFrontEnd kserver = KServerFrontEnd.instance();
         if (!kserver.isLocal()) {
             context.assertLoopbackClient();
         }
         isNailgun = true;
+        Main.context = context;
         if (context.getArgs().length >= 1) {
             String[] args2 = Arrays.copyOfRange(context.getArgs(), 1, context.getArgs().length);
             int result = kserver.run(context.getArgs()[0], args2, new File(context.getWorkingDirectory()), (Map) context.getEnv(), startTime);
-            System.exit(result);
+            context.exit(result);
```

Comment:
What's the difference between `Main.exit` and `context.texit`?

Reply:
context.exit can be called if you are 100% sure that you are running nailgun. It's probably best to be consistent though; I will fix this.